src/careamics/dataset_ng/patch_extractor/patch_extractor_factory.py

```diff
@@ -95,6 +96,48 @@ def create_ome_zarr_extractor(
     return PatchExtractor(image_stacks)
 
 
+# CZI case
+def create_czi_extractor(
+    source: Sequence[Path],
+    axes: str,
+) -> PatchExtractor[CziImageStack]:
+    """
+    Create a patch extractor from a sequence of CZI files.
+
+    If the CZI files contain multiple scenes, one patch extractor will be created for
+    each scene.
+
+    Parameters
+    ----------
+    source: sequence of Path
+        The source files for the data.
+    axes: str
+        Specifies which axes of the data to use and how.
+        If this string ends with "ZYX", the data will consist of 3-D patches.
+        If the data has no Z axis but a T axis, the T axis will be used as
+        3rd dimension.
+        If the string does not end with "ZYX", the data will consist of 2-D patches.
+
+    Returns
+    -------
+    PatchExtractor
+    """
+    depth_axis: Literal["none", "Z", "T", "auto"] = "none"
+    if axes.endswith("TYX"):
+        depth_axis = "T"
+    elif "Z" in axes:
+        depth_axis = "auto"
```

Comment:
Joining the previous comment about `depth_axis`, "Z" is not used here, so it may not be necessary?

Reply:
I agree the current handling was confusing and to some degree nontransparent.

I now removed the `"auto"` option for `"depth_axis"`. The new default is `"none"`, yielding 2-D patches. 3-D patches must be requested explicitly by either setting `depth_axis="Z"` or `depth_axis="T"`.

In terms of the `create_czi_extractor` factory, that means that `axes="ZYX"` and `axes="TYX"` will yield 3-D patches, everything else 2-D patches.

I think this is cleaner.